### PR TITLE
Tarred audio support in ASR data layer

### PR DESCRIPTION
**Pull request by redoctopus**
Adding support for loading ASR datasets with tarred audio, using the WebDataset library. Includes a script to convert existing datasets to a TarredAudioToTextDataLayer-accepted format.

### CODE REVIEW

File: nemo/collections/asr/data_layer.py [hunk 1]
```diff
@@ -340,6 +347,234 @@ def data_iterator(self):
...
+        self,
+        audio_tar_filepaths,
+        manifest_filepath,
+        labels,
+        batch_size,
+        sample_rate=16000,
+        int_values=False,
+        bos_id=None,
+        eos_id=None,
+        pad_id=None,
```
Comment (blisc) [inline nemo/collections/asr/data_layer.py:445]: Can you add bos_id, and pad_id to class docstring?

File: scripts/convert_to_tarred_audio_dataset.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+#
+# This script converts an existing audio dataset with a manifest to
+# a tarred and sharded audio dataset that can be read by the
+# TarredAudioToTextDataLayer.
+
+import argparse
+import json
+import os
+import random
+import tarfile
```
Comment (blisc) [inline scripts/convert_to_tarred_audio_dataset.py:23]: is tarfile in our requirements anywhere?

File: tests/unit/test_unit_asr.py [hunk 3]
```diff
@@ -254,6 +254,47 @@ def test_kaldi_dataloader(self):
...
 
+    @pytest.mark.unit
+    def test_tarred_dataloader(self):
+        batch_size = 4
+        manifest_path = os.path.abspath(
+            os.path.join(os.path.dirname(__file__), '../data/asr/tarred_an4/tarred_audio_manifest.json')
+        )
+
+        # Test loading a single tarball
+        tarpath = os.path.abspath(os.path.join(os.path.dirname(__file__), '../data/asr/tarred_an4/audio_0.tar'))
```
Comment (blisc) [inline tests/unit/test_unit_asr.py:265]: When is this data loaded? Is this what was added to the tests/data/asr.tar.gz file?

File: nemo/collections/asr/data_layer.py (hunk 1 above)
Comment (redoctopus) [reply nemo/collections/asr/data_layer.py:457]: Done. Also added them to the AudioToTextDataLayer docstring.

File: scripts/convert_to_tarred_audio_dataset.py (hunk 2 above)
Comment (redoctopus) [reply scripts/convert_to_tarred_audio_dataset.py:23]: Nope, added now! (To the ASR requirements--reasoning is that this script is specifically for converting ASR datasets.)

File: tests/unit/test_unit_asr.py (hunk 3 above)
Comment (redoctopus) [reply tests/unit/test_unit_asr.py:265]: Yes, I added a `tarred_an4` directory. It contains files `audio_{0..3}.tar` as well as the manifest used in the tests.

File: scripts/convert_to_tarred_audio_dataset.py (hunk 2 above)
Comment (redoctopus) [reply scripts/convert_to_tarred_audio_dataset.py:23]: ...Actually, it looks like it's part of the standard library. Going to remove it.

**Pull request merged**